### PR TITLE
Support TON Connect standard  CHA-42

### PR DESCRIPTION
**Pull request by lmangall**
https://linear.app/chatpay/issue/CHA-233/support-ton-connect-standard

### CODE REVIEW

**Comment by 552020** [comment]
Branch has conflicts.

**Review by 552020 — CHANGES_REQUESTED** [review]
Conflicts

**Comment by lmangall** [comment]
Solved

File: manifest.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+	"name": "ChatPay",
+	"description": "Earn money from your Telegram chats by bundling them into AI training datasets.",
```
Comment (552020) [inline manifest.json:3]: Has this description been approved? I.e. is a starndard one we have?

File: manifest.json
Comment (552020) [inline manifest.json:1]: For what is this manifest.json?

File: src/components/Word.tsx [hunk 2]
```diff
@@ -12,53 +13,51 @@ const Word: React.FC = () => {
...
-          description=''
-          header='Balance'
-          subheader={`${balance} $WORDS`}
-          type='inline'
-        >
-          <React.Fragment key='.0'></React.Fragment>
-        </Banner>
-      </div>
+    <TonConnectUIProvider
+      manifestUrl='https://yourappurl.com/tonconnect-manifest.json'
```
Comment (552020) [inline src/components/Word.tsx:17]: Here we need our app url, not 'yourappurl.com'!

**Review by 552020 — CHANGES_REQUESTED** [review]
Good job, still some question and a minor change request.

File: manifest.json (hunk 1 above)
Comment (lmangall) [reply manifest.json:3]: It is

File: manifest.json
Comment (lmangall) [reply manifest.json:1]: It's in their doc:
https://docs.ton.org/develop/dapps/ton-connect/web

Maybe we should open an issue to improve it in the future

File: src/components/Word.tsx (hunk 2 above)
Comment (lmangall) [reply src/components/Word.tsx:17]: We don't have a logo online, I just set the domain name to the vercel deployment, I guess as soon as we put a favicon for the website we can set the link here

**Review by 552020 — APPROVED** [review]
LGTM

File: src/components/Word.tsx (hunk 2 above)
Comment (552020) [reply src/components/Word.tsx:17]: This was about ot the logo online it was about having the our url there pointing to tonconnect-manifest.json

**Pull request merged**